### ISSUE TITLE
Extremely long class list prevents exiting QuPath

### ISSUE THREAD

**Issue by petebankhead** [qupath/qupath#626]
## Bug report

**Describe the bug**
QuPath has major problems if the list of available classifications is too long.

See https://forum.image.sc/t/extremely-long-class-list-prevents-exiting-qupath/44381 for details.

**Expected behavior**
QuPath closes elegantly, and retrieves the class list when restarted.

**Additional context**
Presumably we hit the [8192 character limit of what can be stored in preferences](https://docs.oracle.com/javase/8/docs/api/java/util/prefs/Preferences.html#MAX_VALUE_LENGTH).

Note that this also impacts byte arrays (since they use Base64 encoding).

Options include:
* Use the user directory instead (but this might not be set)
* Encode the preferences in a different way, e.g. across multiple entries if needed

**Comment by petebankhead** [comment]
A script like the following can be used to replicate the problem:
```groovy
guiscript=true

def pathClasses = []
for (int i = 0; i < 500; i++) {
    pathClasses << getPathClass('Another class ' + i)
}

getQuPath().getAvailablePathClasses().clear()
getQuPath().getAvailablePathClasses().addAll(pathClasses)
```

**Comment by petebankhead** [comment]
The fix will only save classification lists to the preferences if they can fit. Otherwise, the classification list can still be retrieved from a project in the future - but won't be stored in the preferences.

**Comment by petebankhead** [comment]
Fixed in https://github.com/qupath/qupath/pull/635 - will be included in v0.3.0.
